core/Plugin/Manager.php

```diff
@@ -516,6 +521,7 @@ public function deactivatePlugin($pluginName)
 
         // execute deactivate() to let the plugin do cleanups
         $this->executePluginDeactivate($pluginName);
+        $this->savePluginDeactivationTime($pluginName);
```

Comment:
@diosmosis bit unrelated probably but could we execute the deactivate logic only if the plugin is actually activated so it can't be triggered with just any random plugin name?

Reply:
eg in `activatePlugin` we do:

```
        $plugins = $this->pluginList->getActivatedPlugins();
        if (in_array($pluginName, $plugins)) {
            // plugin is already activated
            return;
        }
```

and could use similar logic